0xProject/0x-monorepo
# Move CLI tests for abi-gen

## Description

All CLI-related tests now live in `test-cli` folder. There are separate subfolders for `python` and `typescript`, which will hopefully make it easier to add language-specific tests down the road. 

`yarn clean`, `yarn build`, and `yarn test` will now clean, build, and test the abi-gen utility AND the related CLI tests. 

```
packages/abi-gen
├── test
│   └── utils_test.ts
├── test-cli
│   ├── expected-output
│   │   ├── python
│   │   │   ├── abi_gen_dummy.py
│   │   │   ├── lib_dummy.py
│   │   │   └── test_lib_dummy.py
│   │   └── typescript
│   │       ├── abi_gen_dummy.ts
│   │       ├── lib_dummy.ts
│   │       └── test_lib_dummy.ts
│   ├── fixtures
│   │   ├── artifacts
│   │   │   ├── AbiGenDummy.json
│   │   │   ├── LibDummy.json
│   │   │   └── TestLibDummy.json
│   │   └── contracts
│   │       ├── AbiGenDummy.sol
│   │       ├── LibDummy.sol
│   │       └── TestLibDummy.sol
│   ├── output (generated files, gitignored)
│   │   ├── python
│   │   │   ├── abi_gen_dummy.py
│   │   │   ├── lib_dummy.py
│   │   │   └── test_lib_dummy.py
│   │   └── typescript
│   │       ├── abi_gen_dummy.ts
│   │       ├── lib_dummy.ts
│   │       └── test_lib_dummy.ts
│   ├── test_typescript (unit tests for generated typescript)
│   │   ├── lib (tsc build output)
│   │   ├── src
│   │   │   ├── artifacts.ts
│   │   │   ├── index.ts
│   │   │   └── wrappers.ts
│   │   └── test
│   │       └── abi_gen_dummy_test.ts
│   └── tsconfig.json
├── tsconfig.json
└── tslint.json

```
Resolves #1925, Resolves #1926

## Testing instructions

The best way to test it is to check it out and run the workflow yourself. Would appreciate feedback on comprehensibility and ease of use. 

Make sure to clean first. 

```
yarn clean && yarn build && yarn test
```

## Types of changes



 * Bug fix (non-breaking change which fixes an issue) 





## Checklist:
